integration_test/index_test.go

```diff
@@ -171,3 +172,57 @@ func TestKrewIndexRemoveForce_nonExisting(t *testing.T) {
 	// --force returns success for non-existing indexes
 	test.Krew("index", "remove", "--force", "non-existing").RunOrFail()
 }
+
+func TestKrewDefaultIndex_notAutomaticallyAdded(t *testing.T) {
+	skipShort(t)
+	test, cleanup := NewTest(t)
+	test = test.WithEnv(constants.EnableMultiIndexSwitch, 1)
+	defer cleanup()
+
+	test.Krew("help").RunOrFail()
+	out, err := test.Krew("search").Run()
+	if err == nil {
+		t.Fatalf("search must've failed without any indexes. output=%s", string(out))
+	}
+	out = test.Krew("index", "list").RunOrFailOutput()
+	if len(lines(out)) > 1 {
+		t.Fatalf("expected no indexes; got output=%q", string(out))
+	}
+}
+
+func TestKrewDefaultIndex_AutoAddedOnInstall(t *testing.T) {
+	skipShort(t)
+	test, cleanup := NewTest(t)
+	test = test.WithEnv(constants.EnableMultiIndexSwitch, 1)
+	defer cleanup()
+
+	test.Krew("install", validPlugin).RunOrFail()
+	ensureIndexListHasDefaultIndex(t, string(test.Krew("index", "list").RunOrFailOutput()))
+}
+
+func TestKrewDefaultIndex_AutoAddedOnUpdate(t *testing.T) {
+	skipShort(t)
+	test, cleanup := NewTest(t)
+	test = test.WithEnv(constants.EnableMultiIndexSwitch, 1)
+	defer cleanup()
+
+	test.Krew("update").RunOrFail()
+	ensureIndexListHasDefaultIndex(t, string(test.Krew("index", "list").RunOrFailOutput()))
+}
+
+func TestKrewDefaultIndex_AutoAddedOnUpgrade(t *testing.T) {
+	skipShort(t)
+	test, cleanup := NewTest(t)
+	test = test.WithEnv(constants.EnableMultiIndexSwitch, 1)
+	defer cleanup()
+
+	test.Krew("upgrade").RunOrFail()
+	ensureIndexListHasDefaultIndex(t, string(test.Krew("index", "list").RunOrFailOutput()))
+}
+
+func ensureIndexListHasDefaultIndex(t *testing.T, output string) {
+	t.Helper()
+	if !regexp.MustCompile(`(?m)^default\b`).MatchString(output) {
```

Comment:
Nit: isn't it better to put this at the package level so that the regular expression is only parsed once instead of for each time this function is called?

Reply:
since it's only part of integration tests, I figured it doesn't matter a whole lot.